### PR TITLE
Port TaroNuke's quick restart feature (ctrl + r) in gameplay

### PR DESCRIPTION
**Pull request by teejusb**
Seems pretty heavily requested. F3+F6+2 works but is unintuitive in general.

### CODE REVIEW

**Comment by natano** [comment]
On every quick restart this message is printed to the log:
```
/////////////////////////////////////////
WARNING: XXX: m_iNumStagesOfThisSong == 2?
/////////////////////////////////////////
```
It is raised by the engine in `GameState::BeginStage()` because it is called twice for the same stage. It would be good to find out if it points out an actual issue that should be fixed or if we can remove the warning.

File: BGAnimations/ScreenGameplay underlay/default.lua [hunk 1]
```diff
@@ -4,7 +4,34 @@ if PREFSMAN:GetPreference("MenuTimer") then
...
+		if event.DeviceInput.button == "DeviceButton_left ctrl" then
+			holdingCtrl = false
+		end
+	end
+end
+
+local t = Def.ActorFrame{
+	Name="GameplayUnderlay",
+	OnCommand=function(self)
+		if ThemePrefs.Get("KeyboardFeatures") and PREFSMAN:GetPreference("EventMode") then
```
Comment (natano) [inline BGAnimations/ScreenGameplay underlay/default.lua]: We probably also want to disable this for courses.

**Review by natano — APPROVED** [review]

File: BGAnimations/ScreenGameplay underlay/default.lua (hunk 1 above)
Comment (teejusb) [reply BGAnimations/ScreenGameplay underlay/default.lua]: Done.

**Comment by teejusb** [comment]
Fixed the warning by doing:
`SCREENMAN:GetTopScreen():SetPrevScreenName("ScreenGameplay"):SetNextScreenName("ScreenGameplay"):begin_backing_out()`

(same thing as restart on missed target)

**Pull request merged**